app/models/payment.rb

```diff
@@ -9,6 +9,8 @@ def self.valid_sources
   end
 
   validates :source, presence: true, inclusion: { in: valid_sources }
-  validates :account, :amount, presence: true
+  validates :account, presence: true
+  validates :amount, presence: true, numericality: true
+  validates :processing_fee, numericality: true, allow_nil: false
```

Comment:
Actually, I realize that may be incorrect. `0.present? == true`. Let me rethink that a bit.